test/api/v3/unit/models/group.test.js

```diff
@@ -434,6 +434,37 @@ describe('Group Model', () => {
       });
     });
 
+    describe('#leaveGroup', () => {
+      it('removes user from group quest', async () => {
+        party.quest.members = {
+          [participatingMember._id]: true,
+          [questLeader._id]: true,
+          [nonParticipatingMember._id]: false,
+          [undecidedMember._id]: null,
+        };
+        party.memberCount = 4;
+        await party.save();
+
+        await party.leave(participatingMember);
+
+        party = await Group.findOne({_id: party._id});
+        expect(party.quest.members).to.eql({
+          [questLeader._id]: true,
+          [nonParticipatingMember._id]: false,
+          [undecidedMember._id]: null,
+        });
+      });
+
+      it('deletes group if user is only member', async () => {
```

Comment:
In fact, for bonus points you could add a test for it not deleting a public guild.